src/App.jsx

```diff
@@ -1,34 +1,71 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import { GlobalStyles } from "./styles/GlobalStyles";
+import { getDate } from "./utils/helpers";
+import { useState } from "react";
+import Wrapper from "./components/Wrapper";
+import Heading from "./components/Heading";
+import DateParagraph from "./components/DateParagraph";
+import Input from "./components/Input";
+import List from "./components/List";
 
 function App() {
-  const [count, setCount] = useState(0)
-
-  return (
-    <div className="App">
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src="/vite.svg" className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://reactjs.org" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </div>
-  )
+	const [tasks, setTasks] = useState([
+		{ id: 1, title: "Fake task from state", done: false },
+		{ id: 2, title: "Fake task from state", done: true },
+	]);
+
+	const changeTaskStatus = (index) => {
+		tasks[index].done = !tasks[index].done;
+		setTasks([...tasks]);
+	};
+
+	const addTask = (task) => {
+		setTasks([task, ...tasks]);
+	};
+
+	const deleteTask = (id) => {
+		const filteredTasks = tasks.filter((task) => task.id !== id);
+		setTasks([...filteredTasks]);
+	};
+
+	const [formValue, setFormValue] = useState({
+		id: crypto.randomUUID(),
+		title: "",
+		done: false,
+	});
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		addTask(formValue);
+		setFormValue({ id: crypto.randomUUID(), title: "", done: false });
+	};
+
+	return (
+		<>
+			<GlobalStyles />
+			<Wrapper>
+				<header>
+					<Heading title="My tasks" />
+					<DateParagraph currentDate={getDate()} />
+				</header>
+				<form onSubmit={handleSubmit}>
+					<Input
+						id="test"
+						type="text"
+						placeholder="+ Add your new task..."
+						value={formValue.title}
+						onChange={(e) =>
```

Comment:
Zadziała, fakt, ale może spróbuj wynieść sobie tę funkcję wyżej i wrzuć ją tutaj jako consta? Plus jeśli mieliście useCallback może zoptymalizuj ją pod kątem rerenderingu jej - chyba, że nie mieliście, to zrób tylko pierwszą część.

Reply:
Trochę poczytałem i zrobiłem ten useCallback, choć szczerze nie wiem czy dobrze, bo trochę intuicyjnie i metodą prób i błędów :)